crates/cli/src/cli.rs

```diff
@@ -1,114 +1,110 @@
-
     let network_items = if is_debug_network {
         vec!["Mainnet", "Sepolia", "Integration", "Local"]
     } else {
         vec!["Mainnet", "Sepolia"]
     };
-
-    let network_index: Option<usize> = Select::with_theme(&ColorfulTheme::default())
+    let network_index = Select::with_theme(&ColorfulTheme::default())
         .items(&network_items)
         .with_prompt("Which network would you like to verify on : ")
         .default(0)
-        .interact_opt()?;
+        .interact_opt()
+        .expect("Aborted at network selection, terminating...")
+        .expect("Aborted at network selection, terminating...");
+
+    // Project type + Path entry
+    let target_type = TargetType::ScarbProject; // by default we assume the user is in a scarb project
+    let is_current_dir_scarb = env::current_dir()?.join("scarb.toml").exists();
+    let path = if is_current_dir_scarb {
+        env::current_dir()?.to_str().unwrap().trim().to_string()
+    } else {
+        // TODO, add TargetType::File path input here
+        Input::<String>::with_theme(&ColorfulTheme::default())
+            .with_prompt("Enter Path to scarb project root:")
+            .interact_text()
+            .expect("Aborted at path input, terminating...")
+            .trim()
+            .to_string()
+    };
+    let utf8_path: Utf8PathBuf = Utf8PathBuf::from(path);
+    if !utf8_path.exists() {
+        panic!("Path does not exist");
+    }
 
-    match network_index {
-        Some(_) => (),
-        None => {
-            println!("Aborted at network selection, terminating...");
-            std::process::exit(1);
+    // Start the whole process
+    let _spinner_style = ProgressStyle::with_template("{prefix:.bold.dim} {spinner} {wide_msg}")
+        .unwrap()
+        .tick_chars("⠁⠂⠄⡀⢀⠠⠐⠈");
+
+    println!(
+        "{} {} Resolving project...",
+        style("[2/3]").bold().dim(),
+        Emoji("🔗", "")
+    );
+
+    // Resolve project
+    let (project_files, project_metadata) = match target_type {
+        TargetType::File => {
+            panic!("Single contract file verification is not yet implemented, please use a scarb project instead.");
         }
-    }
+        TargetType::ScarbProject => {
+            let (local_scarb_version, local_cairo_version) = detect_local_tools();
+            // TODO: do a first pass to find all the contracts in the project
+            // For now we keep using the hardcoded value in the scarb.toml file
 
-    let re = Regex::new(r"^0x[a-fA-F0-9]{64}$").unwrap();
+            resolver::resolve_scarb(utf8_path.clone(), local_cairo_version, local_scarb_version)?
+        }
+    };
 
+    // TODO: try to calculate the class hash automatically later after contract selection?
+    // println!(
+    //     "{} {} Calculating class hash...",
+    //     style("[x/x]").bold().dim(),
+    //     Emoji("🔍  ", "")
+    // );
+    let re = Regex::new(r"^0x[a-fA-F0-9]{64}$").unwrap();
```

Comment:
I think you should add a step 3 here, since this is the collection of information. Or maybe you should make resolving step 1, then step 2 is the colllection of information

Reply:
I'll do:
- 1. Getting project information (input)
- 2. Resolving project (auto)
- 3. Getting verification information: class hash and network (input, and we can try to make class hash calculation auto later)
- 4. Verify (auto)